#analysis for bam file generated by tophat2
library(edgeR)
library(Rsubread)
library(org.Hs.eg.db)

bam_files_path = '/n/scratch2/cc400/align_star/bam_files'
output_path = '~/Documents/workspace/phospho_network/RNAseq/processed_data/star/'
gtf_file   = '/n/scratch2/cc400/align_star/Homo_sapiens.GRCh38.84.gtf'
genome_file = '/n/scratch2/cc400/align_star/Homo_sapiens.GRCh38.dna.primary_assembly.fa'

bam_files <- paste(bam_files_path,grep('.bam',list.files(bam_files_path),value = T,fixed = T),sep = '/')
sample_name  <- gsub('.+/(.+)_qcAligned.sortedByCoord.out.bam','\\1',bam_files)

fc <- featureCounts(files=bam_files,annot.inbuilt="hg38",genome = genome_file,isPairedEnd = T)

x <- DGEList(counts=fc$counts, genes=fc$annotation[,c("GeneID","Length")])
x_rpkm <- rpkm(x,x$genes$Length)
colnames(x_rpkm) <- sample_name
rownames(x_rpkm) <- select(org.Hs.eg.db, rownames(x_rpkm),"SYMBOL")[,2]
write.csv(x_rpkm,paste(output_path,'rpkm.csv',sep = ''))

count_table <- fc$counts
write.csv(count_table,paste(output_path,'count_table.csv',sep = ''))

annotation_table <- fc$annotation
write.csv(annotation_table,paste(output_path,'annotation_table.csv',sep = ''))

stats_table = fc$stat
write.csv(stats_table,paste(output_path,'stats_table.csv',sep = ''))
